v3/src/hooks/use-outside-pointer-down.ts

```diff
@@ -0,0 +1,61 @@
+/* Patterned after Chakra's useOutsideClick hook */
+
+import { useEffect } from "react"
+import { useCallbackRef } from "./use-callback-ref"
+
+interface IProps {
+  /**
+   * Whether the hook is enabled
+   */
+  enabled?: boolean
+  /**
+   * The reference to a DOM element.
+   */
+  ref: React.RefObject<HTMLElement>
+  /**
+   * Function invoked when a click is triggered outside the referenced element.
+   */
+  handler?: (e: Event) => void
+}
+
+/**
+ * Example, used in components like Dialogs and Popovers, so they can close
+ * when a user clicks outside them.
+ */
+export function useOutsidePointerDown({ ref, handler, enabled = true }: IProps) {
+  const savedHandler = useCallbackRef(handler)
+
+  useEffect(() => {
+    if (!enabled) return
+
+    const onPointerDown: any = (e: PointerEvent) => {
+      if (isValidEvent(e, ref)) {
+        savedHandler(e)
+      }
+    }
+
+    // listen for pointerdown because D3 drag handlers intercept subsequent mouse events
+    const doc = getOwnerDocument(ref.current)
+    doc.addEventListener("pointerdown", onPointerDown, true)
```

Comment:
Nice.  So the mouse and other events from the original were extraneous.

Reply:
In this particular instance, the issue was that the mouse and other events were never getting to the hook in the case we cared about, so by definition the handlers for them were extraneous. As near as I can tell, most of the subtle logic in the Chakra hook was to detect the down event but then only call the callback (e.g. close the dialog) when the up event occurred. I don't see why we care about that distinction or whether we close on a `pointerdown` or other event. `pointerdown` happens to be the first event sent, however, and the only one that isn't swallowed by D3's drag handlers, so that's the one we use.